luxtronik/datatypes.py

```diff
@@ -194,55 +205,28 @@ class Errorcode(Base):
     datatype_class = "errorcode"
 
 
-class Kelvin(Base):
+class Kelvin(ScalingBase):
     """Kelvin datatype, converts from and to Kelvin."""
 
     datatype_class = "temperature"
     datatype_unit = "K"
+    factor = 0.1
 
-    @classmethod
-    def from_heatpump(cls, value):
-        if value is None:
-            return None
-        return value / 10
 
-    @classmethod
-    def to_heatpump(cls, value):
-        return int(float(value) * 10)
-
-
-class Pressure(Base):
+class Pressure(ScalingBase):
     """Pressure datatype, converts from and to Pressure."""
 
     datatype_class = "pressure"
     datatype_unit = "bar"
+    factor = 0.01
 
-    @classmethod
-    def from_heatpump(cls, value):
-        if value is None:
-            return None
-        return value / 100
 
-    @classmethod
-    def to_heatpump(cls, value):
-        return int(value * 100)
-
-
-class Percent(Base):
+class Percent(ScalingBase):
```

Comment:
> Maybe we should rename then to `Percent10` and `Percent1` to indicate the (inverse) scaling factor in the name?

Good suggestion. Do you want to do this here in this pull request, since this is than related to the scaling factor?

Reply:
I don't think that would fit nicely here, because this will involve touching (almost) all the files.

I was also thinking of putting the scaling factor into the constructor, e.g., `Percent("name", 0.1)`, but this is not possible, since the scaling factor has to be a class variable. In C++ one could use template parameters like `Percent<0.1>("name")` but I don't think that this is possible in Python.

This would also apply to all children of `SelectionBase`, since these are used only once in `parameters.py` or `calculations.py` and it seems overkill to have an own class (for only one parameter).